syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.miotech.charmander.dmp.protobuf";
option java_outer_classname = "DmpCharmander";

import "dmp.proto";
import "search.proto";

service DmpCharmanderService {

    //  status: Ready
    //  priority: high
    //  external interface
    rpc queryAssetMarketDataSeries (com.mioying.data.models.protobuf.AssetDataRequest) returns (com.mioying.data.models.protobuf.AssetMarketDataSeries) {
    }

    //  status: Ready
    //  priority:
    //  internal interface
    rpc search (SearchRequest) returns (com.mioying.data.models.protobuf.SearchResult) {
    }

    //  status: Need to test
    //  priority: high
    //  external interface
    rpc retrieveAssetId (com.mioying.data.models.protobuf.RetrieveAssetIdRequest) returns (com.mioying.data.models.protobuf.RetrieveAssetIdResult) {
    }

    //  status: Ready
    //  priority: low
    //  external interface
    rpc retrieveCurrencyRate (com.mioying.data.models.protobuf.RetrieveCurrencyRateRequest) returns (com.mioying.data.models.protobuf.RetrieveCurrencyRateResult) {
    }

    //  status: Ready
    //  priority: high
    //  external interface
    rpc retrieveHistoryCurrencyRate (com.mioying.data.models.protobuf.RetrieveHistoryCurrencyRateRequest) returns (com.mioying.data.models.protobuf.RetrieveHistoryCurrencyRateResult) {
    }

    //  status: Ready
    //  priority: high
    //  external interface
    rpc queryAsset (com.mioying.data.models.protobuf.AssetRequest) returns (com.mioying.data.models.protobuf.AssetResult) {
    }

    //  status: Not Support
    //  priority:
    //  external interface
    rpc queryTrashAsset (com.mioying.data.models.protobuf.AssetRequest) returns (com.mioying.data.models.protobuf.AssetResult) {
    }

    //  status: Ready
    //  priority: high
    //  external interface
    rpc queryLastAssetMarketDataSeries (com.mioying.data.models.protobuf.AssetDataRequest) returns (com.mioying.data.models.protobuf.AssetMarketDataSeries) {
    }

    //  status: Ready
    //  priority: medium
    //  external interface
    rpc createUserProduct (com.mioying.data.models.protobuf.CreateUserProductRequest) returns (com.mioying.data.models.protobuf.CreateUserProductResult) {
    }

    //  status: Need to fix
    //  priority: low
    //  external interface
    rpc updateUserProduct (com.mioying.data.models.protobuf.UpdateUserProductRequest) returns (com.mioying.data.models.protobuf.UpdateUserProductResult) {
    }

    //  status: Ready
    //  priority: low
    //  external interface
    rpc deleteUserProduct (com.mioying.data.models.protobuf.DeleteUserProductRequest) returns (com.mioying.data.models.protobuf.DeleteUserProductResult) {
    }

    //  status: Need to test
    //  priority: medium
    //  external interface
    rpc checkUserProductExist (com.mioying.data.models.protobuf.CheckUserProductRequest) returns (com.mioying.data.models.protobuf.CheckUserProductResult) {
    }

    //  status: Need to fix
    //  priority: low
    //  external interface
    rpc updateAssetMarketDataSeries (com.mioying.data.models.protobuf.UpdateAssetMarketDataRequest) returns (com.mioying.data.models.protobuf.UpdateAssetMarketDataResult) {
    }

    //  status: Ready
    //  priority: low
    //  external interface
    rpc listAllUserProduct (com.mioying.data.models.protobuf.ListAllUserProductRequest) returns (com.mioying.data.models.protobuf.ListAllUserProductResult) {
    }

    //  status: Not Support
    //  priority:
    //  external interface
    rpc refreshAssetData (com.mioying.data.models.protobuf.RefreshDataRequest) returns (com.mioying.data.models.protobuf.RefreshDataResult) {
    }

    //  status: Ready
    //  priority: low
    //  external interface
    rpc listAllCountryInfo (com.mioying.data.models.protobuf.ListAllRequest) returns (com.mioying.data.models.protobuf.CountryInfoResult) {
    }

    //  status: Ready
    //  priority: low
    //  external interface
    rpc listAllExchangeInfo (com.mioying.data.models.protobuf.ListAllRequest) returns (com.mioying.data.models.protobuf.ExchangeInfoResult) {
    }
}